File: app/src/main/java/com/overswayit/plesnisavezsrbije/viewmodels/ClubViewModel.java
package com.overswayit.plesnisavezsrbije.viewmodels;

import android.app.Application;

import com.overswayit.plesnisavezsrbije.models.Club;
import com.overswayit.plesnisavezsrbije.repository.ClubRepository;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

/**
 * Created by lazarristic on 23/04/2019.
 * Copyright (c) 2019 PlesniSavezSrbije. All rights reserved.
 */
public class ClubViewModel extends AndroidViewModel {
    private ClubRepository clubRepository;

    public ClubViewModel(int id, @NonNull Application application) {
        super(application);
        this.clubRepository = new ClubRepository(id, application);
    }

    public LiveData<Club> getClub() {
        return clubRepository.getClubLiveData();
    }
}

File: app/src/main/java/com/overswayit/plesnisavezsrbije/viewmodels/ClubViewModel.kt
package com.overswayit.plesnisavezsrbije.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData
import androidx.lifecycle.viewModelScope
import com.overswayit.plesnisavezsrbije.models.Club
import com.overswayit.plesnisavezsrbije.repository.ClubsRepository
import kotlinx.coroutines.launch

/**
 * Created by lazarristic on 23/04/2019.
 * Copyright (c) 2019 PlesniSavezSrbije. All rights reserved.
 */
class ClubViewModel(id: Int, application: Application) : AndroidViewModel(application) {
    private val clubsRepository = ClubsRepository(application)
    private val observableClub = MediatorLiveData<Club>()

    init {
        viewModelScope.launch {
            observableClub.addSource(clubsRepository.findById(id), observableClub::setValue)
        }
    }

    val club: LiveData<Club>
        get() = observableClub
}
